feat: add signed url uploads with some initial tests

JanStevens opened the pull request:
Hi,

Since we desperately need #8 I've taken a stab at it. I know somebody already attempted it in #3 but we are a year later so lets try  this again. I've mainly looked at how `getSignedUrl` was implemented and applied it to all supported drivers

JanStevens committed:
feat: add signed url uploads with some initial tests

thetutlage commented:
We are missing tests for the gcs driver

JanStevens commented:
> We are missing tests for the gcs driver

Hi, yes indeed! Do you have any instructions on setting up a local test environment for all the different drivers? 🤔
The tests are not really running because of missing env variables

Regards,

thetutlage commented:
> Hi, yes indeed! Do you have any instructions on setting up a local test environment for all the different drivers?

I have created real accounts with all these services and use the API Keys for testing. If you do not want to create accounts, then maybe you can add the tests (doing your best) and then I can test them locally myself.

Here's the `.env` file I use

```env
S3_SERVICE=do
S3_BUCKET=testing-flydrive
S3_ACCESS_KEY=
S3_ACCESS_SECRET=
S3_ENDPOINT=https://sgp1.digitaloceanspaces.com
S3_REGION=sgp1
S3_CDN_URL=https://testing-flydrive.sgp1.cdn.digitaloceanspaces.com

S3_SERVICE=r2
S3_BUCKET=testing-flydrive
S3_ACCESS_KEY=
S3_ACCESS_SECRET=
S3_ENDPOINT=https://<id>.r2.cloudflarestorage.com
S3_REGION=auto
S3_CDN_URL=https://pub-<id>.r2.dev

GCS_KEY=file:./gkey.json
GCS_BUCKET=drive-gcs
GCS_FINE_GRAINED_ACL_BUCKET=drive-gcs-no-uniform-acl
```

JanStevens committed:
chore: add missing tests

JanStevens commented:
@thetutlage That would be very helpful! Thanks 🙏 
I've added additional tests for gcs, rechecked the tests for s3 and added additional tests for the fs driver aswel.

If you could run it locally and let me know the results I can iterate on it!

thetutlage commented:
The implementation looks fine to me. However, tests needs some tweaks to pass. I will merge the PR and make needed adjustments

thetutlage commented:
Made a few changes in https://github.com/flydrive-js/core/commit/6dffc02b77f4a08c1b9c742004b73b3345724a57#diff-4ef7ea9a3370783b81f70f863234a280fcbd47c24d355ec903fab11d120e6e81
